lib/models/BucketLoggingStatus.ts

```diff
@@ -0,0 +1,193 @@
+            };
+        }
+
+        if (!parsed) {
+            return {
+                error: { arsenalError: errors.MalformedXML, details: 'request xml is undefined or empty' },
+            };
+        }
+
+        if (!parsed.BucketLoggingStatus) {
+            return {
+                error: { arsenalError: errors.MalformedXML, details: 'missing BucketLoggingStatus root' },
+            };
+        }
+
+        let loggingEnabled: LoggingEnabled | undefined = undefined;
+        if (parsed.BucketLoggingStatus.LoggingEnabled) {
+            const loggingEnabledData = parsed.BucketLoggingStatus.LoggingEnabled[0];
+
+            if (
+                !Object.prototype.hasOwnProperty.call(loggingEnabledData, 'TargetBucket') ||
+                loggingEnabledData.TargetBucket === null ||
+                loggingEnabledData.TargetBucket === undefined
+            ) {
+                return {
+                    error: {
+                        arsenalError: errors.MalformedXML,
+                        details: 'missing TargetBucket field in LoggingEnabled',
+                    },
+                };
+            } else if (loggingEnabledData.TargetBucket[0].length < bucket_logging_status_target_bucket_min_length) {
+                return {
+                    error: {
+                        arsenalError: errors.InvalidBucketName,
+                        details: `TargetBucket field length < ${bucket_logging_status_target_bucket_min_length}`,
+                    },
+                };
+            } else if (loggingEnabledData.TargetBucket[0].length > bucket_logging_status_target_bucket_max_length) {
+                return {
+                    error: {
+                        arsenalError: errors.InvalidBucketName,
+                        details: `TargetBucket field length > ${bucket_logging_status_target_bucket_max_length}`,
+                    },
+                };
+            }
+
+            if (
+                !Object.prototype.hasOwnProperty.call(loggingEnabledData, 'TargetPrefix') ||
+                loggingEnabledData.TargetPrefix === null ||
+                loggingEnabledData.TargetPrefix === undefined
+            ) {
+                return {
+                    error: {
+                        arsenalError: errors.MalformedXML,
+                        details: 'missing TargetPrefix field in LoggingEnabled',
+                    },
+                };
+            } else if (loggingEnabledData.TargetPrefix[0].length > bucket_logging_status_target_prefix_max_length) {
+                return {
+                    error: {
+                        arsenalError: errorInstances.InvalidArgument.
+                            customizeDescription(
+                                `Field exceeds ${bucket_logging_status_target_prefix_max_length} bytes`).
+                            addMetadataEntry('TargetPrefix', loggingEnabledData.TargetPrefix[0]),
```

Comment:
Dot should be at the beginning of the line, not the end.

Weird the formatter or linter did not catch that

```suggestion
                        arsenalError: errorInstances.InvalidArgument
                            .customizeDescription(
                                `Field exceeds ${bucket_logging_status_target_prefix_max_length} bytes`)
                            .addMetadataEntry('TargetPrefix', loggingEnabledData.TargetPrefix[0]),
```

Reply:
done